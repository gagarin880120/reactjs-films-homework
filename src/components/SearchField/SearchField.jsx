import React from 'react';
import styles from './SearchField.module.scss';

export default function SearchField() {
  return (
    <input className={styles.searchInput} type="search" placeholder="the jungle book"/>
  )
}
