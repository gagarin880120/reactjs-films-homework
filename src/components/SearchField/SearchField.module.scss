.searchInput {
  width: 300px;
  height: 38px;
  border: 2px solid #fff;
  border-radius: 5px;
  background: transparent;
  color: #fff;
  font-size: 16px;
  padding-left: 3%;
  outline: 0;
}

.searchInput::placeholder {
  color: #fff;
  font-size: 16px;
  padding-left: 2%;
}
