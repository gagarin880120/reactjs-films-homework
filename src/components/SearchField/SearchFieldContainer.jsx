import React from 'react';
import { connect } from 'react-redux';
import { getResults } from '../../redux/actions';
import SearchField from './SearchField';

export function SearchFieldContainer({ onSearch }) {
  return (
    <SearchField
      onSearch={onSearch}
    />
  )
}

export const mapDispatchToProps = (dispatch) => {
  return {
    onSearch(query) {
      if(query) {
        dispatch(getResults(query));
      }
    }
  }
}

export default connect(null, mapDispatchToProps)(SearchFieldContainer)
