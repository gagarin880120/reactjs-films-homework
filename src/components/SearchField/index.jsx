import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { searchResultsAction } from '../../redux/actions';
import SearchField from './SearchField';

export default function SearchFieldContainer() {
  const dispatch = useDispatch();
  const [query, setQuery] = useState('');

  function getResults(str) {
    return () => {
      const url = new URL('https://api.themoviedb.org/3/search/movie');
      url.searchParams.set('api_key', '306b98213f954f1d07d1d09517898f10');
      url.searchParams.set('language', 'en-US');
      url.searchParams.set('query', str);
      url.searchParams.set('page', '1');
      url.searchParams.set('include_adult', 'false');
      url.searchParams.set('video', 'true');
      fetch(url)
        .then(res => res.json())
        .then(data => dispatch(searchResultsAction(data.results)))
        .catch(e => console.log(e))
    }
  }

  function onKeyDownHandler(e) {
    if (query) {
      if (e.key ==='Enter') {
        return getResults(query)()
      }
    }
  }

  return (
    <SearchField
      query={query}
      setQuery={setQuery}
      onKeyDownHandler={onKeyDownHandler}
    />
  )
}
