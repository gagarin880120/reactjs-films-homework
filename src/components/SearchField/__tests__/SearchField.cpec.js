import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import SearchField from '../SearchField';

test('SearchField component should render without crashing', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<SearchField />);
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
