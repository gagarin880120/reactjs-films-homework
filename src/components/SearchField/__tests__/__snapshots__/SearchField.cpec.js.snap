// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField component should render without crashing 1`] = `
<input
  className="searchInput"
  placeholder="the jungle book"
  type="search"
/>
`;
